go/examples/webhook-server/main.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+
+	tlsigning "github.com/Truelayer/truelayer-signing/go"
+	"github.com/gregjones/httpcache"
+)
+
+func main() {
+	tp := httpcache.NewMemoryCacheTransport()
+	client := http.Client{Transport: tp}
+
+	http.HandleFunc("/hook/d7a2c49d-110a-4ed2-a07d-8fdb3ea6424b", handler(&client))
+
+	log.Println("Starting server on: 7000")
+
+	log.Fatal(http.ListenAndServe(":7000", nil))
+}
+
+func handler(client *http.Client) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPost {
+			http.Error(w, "Method is not supported.", http.StatusNotFound)
+			return
+		}
+
+		err := verifyHook(client, r)
+		if err == nil {
+			w.WriteHeader(http.StatusAccepted)
+		} else {
+			log.Println(err)
+			w.WriteHeader(http.StatusUnauthorized)
+		}
+	}
+}
+
+func verifyHook(client *http.Client, r *http.Request) error {
+	tlSignature := r.Header.Get("Tl-Signature")
+	if len(tlSignature) == 0 {
+		return fmt.Errorf("missing Tl-Signature header")
+	}
+
+	jwsHeader, err := tlsigning.ExtractJwsHeader(tlSignature)
+	if err != nil {
+		return fmt.Errorf("jku missing")
+	}
+
+	defer r.Body.Close()
+	webhookBody, err := io.ReadAll(r.Body)
+	if err != nil {
+		return fmt.Errorf("webhook body missing")
+	}
+
+	// ensure jku is an expected TrueLayer url
+	if jwsHeader.Jku != "https://webhooks.truelayer.com/.well-known/jwks" && jwsHeader.Jku != "https://webhooks.truelayer-sandbox.com/.well-known/jwks" {
+		return fmt.Errorf("unpermitted jku %s", jwsHeader.Jku)
+	}
+
+	// fetch jwks (cached according to cache-control headers)
+	resp, err := client.Get(jwsHeader.Jku)
+	if err != nil {
+		return fmt.Errorf("jku missing")
+	}
+	defer resp.Body.Close()
+	jwks, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("jwks missing")
+	}
```

Comment:
does it make sense to provide this as a function within the library?

Reply:
No http fetching is explicitly out of scope for these libraries, it's a rabbit hole we don't want to maintain here